chord_metadata_service/restapi/api_views.py

```diff
@@ -452,3 +369,48 @@ def sort_numeric_values_into_bins(values: dict, bin_size: int = 10, threshold: i
         values_in_bins[f"{bin_key}"] = keys_sum
     # remove data if count < 5
     return {k: v for k, v in values_in_bins.items() if v > threshold}
+
+
+def stats_for_field(model, field: str):
+    # values() restrict the table of results to this COLUMN
+    # annotate() creates a `total` column for the aggregation
+    # Count() aggregates the results by performing a GROUP BY on the field
+    query_set = model.objects.all().values(field).annotate(total=Count(field))
+    stats = dict()
+    for item in query_set:
+        key = item[field]
+        if key is None:
+            continue
+
+        key = key.strip()
+        if key == "":
+            continue
+
+        stats[key] = item["total"]
+    return stats
+
+
+def get_field_bins(model, field, bin_size):
+    # creates a new field "binned" by substracting the modulo by bin size to
+    # the value which requires binning (e.g. 28 => 28 - 28 % 10 = 20)
+    # cast to integer to avoid numbers such as 60.00 if that was a decimal,
+    # and aggregate over this value.
+    query_set = model.objects.all().annotate(
+        binned=Cast(
+            F(field) - Func(F(field), bin_size, function="MOD"),
+            IntegerField()
+        )
+    ).values('binned').annotate(total=Count('binned'))
+    stats = {item['binned']: item['total'] for item in query_set}
+    return stats
+
+
+def compute_binned_ages(bin_size):
+    a = pheno_models.Individual.objects.filter(age_numeric__isnull=True).values('age')
+    binned_ages = []
+    for r in a.iterator():  # reduce memory footprint (no caching)
+        if r["age"] is None:
+            continue
+        age = parse_individual_age(r["age"])
+        binned_ages.append(age - age % OVERVIEW_AGE_BIN_SIZE)
+    return binned_ages
```

Comment:
Might be better to have them nested within the response function, or have a utility function set file for it, as then the file would have a better logical arrangement that each function as an api response function and not an util function